### ISSUE TITLE
Simple visualization without an expected result

### ISSUE THREAD

**Issue by FiiScho** [aHapBean/PCP-MAE#2]
When I use the prompt of `Simple visualization`, I run with 
`$ python main_vis.py --config cfgs/pretrain/modlenet40.yaml --exp_name final_vis --ckpts experiments/modlenet40/pretrain/exp/ckpt-epoch-300.pth --test`
, but it produces only two files including a copied config.yaml and *.log, and otherwise, there is no other files or the result of visualization

Could someone help me with solving the problem. Thanks a lot.

**Comment by aHapBean** [comment]
Hello,

This is due to the fact that there is only a `base.yaml` file in the `./cfgs/pretrain` directory. I have conducted a test using the following command:

```bash
python main_vis.py --config cfgs/pretrain/base.yaml --exp_name final_vis \
--ckpts ./experiments/final_ckpt_collected/Pre-train/PCP-MAE-300.pth --test
```

The result was satisfactory (in the directory `./vis`). Note the centers used for reconstruction are predicted embeddings, but the centers used for visualization are ground truth centers.

**Comment by FiiScho** [comment]
> Hello,
> 
> This is due to the fact that there is only a `base.yaml` file in the `./cfgs/pretrain` directory. I have conducted a test using the following command:
> 
> ```shell
> python main_vis.py --config cfgs/pretrain/base.yaml --exp_name final_vis \
> --ckpts ./experiments/final_ckpt_collected/Pre-train/PCP-MAE-300.pth --test
> ```
> 
> The result was satisfactory (in the directory `./vis`). Note the centers used for reconstruction are predicted embeddings, but the centers used for visualization are ground truth centers.

Thanks for your answer. Does it mean that the file `main_vis.py` could only be used to visualize the dataset of `ShapeNet` ? And If I want to visualize other datasets, I need to write the corresponding code for it myself?

**Comment by aHapBean** [comment]
Yes, but you might not need to write the code from scratch. You only need to modify the configuration file and adjust ./runner.py accordingly to support ModelNet 40. Please note that the model is pre-trained on ShapeNet, so the reconstruction results on ModelNet 40 may not be optimal.

**Comment by FiiScho** [comment]
> Yes, but you might not need to write the code from scratch. You only need to modify the configuration file and adjust ./runner.py accordingly to support ModelNet 40. Please note that the model is pre-trained on ShapeNet, so the reconstruction results on ModelNet 40 may not be optimal.

Thank you very much.

**Comment by FiiScho** [comment]
> Hello,
> 
> This is due to the fact that there is only a `base.yaml` file in the `./cfgs/pretrain` directory. I have conducted a test using the following command:
> 
> ```shell
> python main_vis.py --config cfgs/pretrain/base.yaml --exp_name final_vis \
> --ckpts ./experiments/final_ckpt_collected/Pre-train/PCP-MAE-300.pth --test
> ```
> 
> The result was satisfactory (in the directory `./vis`). Note the centers used for reconstruction are predicted embeddings, but the centers used for visualization are ground truth centers.

Thanks for your help. I have got the satisfactory result for the first airplane model of `ModelNet40`. And now I want to ask you that how could I get more points for prediction. The number of the points displayed in the bottom picture is 2048, but I cannot find where the number of points for prediction is set in order to get more points. Could you answer me? Thank you again.

![airplane](https://github.com/user-attachments/assets/b0b55b50-ce97-41cc-ab9b-6615b19c8ba3)

**Comment by aHapBean** [comment]
You can change the parameters `npoints`, `group_size`, and `num_group` in the config file. Note that the visualization you obtained may not be seen as satisfactory (in the picture provided by you) because the reconstruction is patch-based. The ground truth centers are used during visualization to retain the rough shape. Since we predict the centers for the point cloud ourselves in PCP-MAE, it may generalize worse to point clouds because our target is more challenging than that of Point-MAE.

**Comment by FiiScho** [comment]
> You can change the parameters `npoints`, `group_size`, and `num_group` in the config file. Note that the visualization you obtained may not be satisfactory because the reconstruction is patch-based. The ground truth centers are used during visualization to retain the rough shape. Since we predict the centers for the point cloud ourselves in PCP-MAE, it may generalize worse to point clouds because our target is more challenging than that of Point-MAE.

Thank you again.
